### PR TITLE
refactor: JOSS revision, round 1

### PR DESCRIPTION
**Pull request by AngryMaciek**
## Description

* add preprocessor directive for `Catch`
* correct typos
* correct power-assignment tests for `H[P]`
* add MPFR `H^30` test to unit tests

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage

### CODE REVIEW

File: .test/unit/test.cpp [hunk 1]
```diff
@@ -1949,7 +1985,7 @@ TEST_CASE( "Hypercomplex: Polynomial lib test", "[unit]" ) {
             REQUIRE( hA[2] == polynomial3 % 2 );
             REQUIRE( hA[3] == polynomial4 % 2 );
         }
-
+
```
Comment (vissarion) [inline .test/unit/test.cpp:1988]: maybe you want to avoid trailing whitespaces

File: .test/unit/test.cpp [hunk 2]
```diff
@@ -1257,6 +1259,39 @@ TEST_CASE( "Hypercomplex: MPFR lib test", "[unit]" ) {
             clear_mpfr_memory();
         }
     }
+
+    SECTION( "Revision test" ) {
+        mpfr_t A[8];
```
Comment (vissarion) [inline .test/unit/test.cpp:1264]: maybe you want to set mpfr precision here to be on the safe side.

File: .test/unit/test.cpp (hunk 1 above)
Comment (AngryMaciek) [reply .test/unit/test.cpp:1988]: Done! Thanks!

File: .test/unit/test.cpp (hunk 2 above)
Comment (AngryMaciek) [reply .test/unit/test.cpp:1264]: This one is for sure OK; Catch2 tests work in a tree-structure; i.e. test cases are evaluated as separate branches with the most-nested cases being leaves. all code from the internal nodes are executed before reaching the leaf. The section above is a subsection of the one in line `521`, right there I set the precision and this gets passed to all branches downstream.

**Pull request merged**